### PR TITLE
WIP: Bootstrap selector [ci skip]

### PR DESCRIPTION
**Pull request by DoriftoShoes**

### CODE REVIEW

**Comment by estee-tew** [comment]
https://stackstorm.slack.com/archives/stackstorm/p1457047501005890

File: scripts/st2_bootstrap_selector.sh [hunk 1]
```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+BASE_PATH="https://raw.githubusercontent.com/StackStorm/st2-packages/master/scripts/st2bootstrap"
```
Comment (dzimine) [inline scripts/st2_bootstrap_selector.sh]: we'll need version instead of `master` here.

File: scripts/st2_bootstrap_selector.sh [hunk 2]
```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
```
Comment (dzimine) [inline scripts/st2_bootstrap_selector.sh]: st2bootstrap.sh for a name?

**Comment by dzimine** [comment]
awesome, hope it works :) 
I assume same mechanics: curl sSL stackstorm.com/bootstrap.sh | sudo su -> st2-packages / master / st2bootstrap.sh -> runs a proper branch (version) of st2bootstrap-xx.sh

File: scripts/st2_bootstrap_selector.sh [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        ;;
+          *)
+                  # unknown option
+          ;;
+      esac
+    done
+
+  if [[ "$RELEASE" == "unstable" ]]; then
+    echo "This script does not support installing from unstable sources!"
+    # XXX: Fix this when st2mistral unstable sources become available!
```
Comment (dzimine) [inline scripts/st2_bootstrap_selector.sh]: it shall be there already @dennybaa has manually put it there. please check.

File: scripts/st2_bootstrap_selector.sh [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    fi
+
+    if [[ "$VERSION" =~ ^[0-9]+\.[0-9]+dev$ ]]; then
+     echo "You're requesting a dev version! Switching to unstable!"
+     RELEASE='unstable'
+    fi
+  fi
+
+  echo "########################################################"
+  echo "          Installing st2 $RELEASE $VERSION              "
```
Comment (dzimine) [inline scripts/st2_bootstrap_selector.sh]: s/st2/StackStorm/

File: scripts/st2_bootstrap_selector.sh [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  TYPE="debs"
+  echo "# Detected Distro is ${DEBTEST}"
+  ST2BOOTSTRAP="${BASE_PATH}-deb.sh"
+elif [[ -n "$RHTEST" ]]; then
+  TYPE="rpms"
+  echo "# Detected Distro is ${RHTEST}"
+  RHMAJVER=`cat /etc/redhat-release | awk '{ print $3}' | cut -d '.' -f1`
+  ST2BOOTSTRAP="${BASE_PATH}-el${RHMAJVER}.sh"
+else
+  echo "Unknown Operating System"
```
Comment (dzimine) [inline scripts/st2_bootstrap_selector.sh]: "Unsupported Operating System\n Please run on Ubuntu/Debian, RHEL7/CentOS7, or RHEL6/CentOS6"

**Comment by dzimine** [comment]
I tried to run it with https://github.com/StackStorm/st2-packages/blob/master/scripts/st2bootstrap-el6.sh on RHEL 6 and it didn't work. @DoriftoShoes knows and will be looking at it.

File: scripts/st2_bootstrap_selector.sh (hunk 1 above)
Comment (lakshmi-kannan) [inline scripts/st2_bootstrap_selector.sh]: That's a different PR I'd think.

File: scripts/st2_bootstrap_selector.sh (hunk 3 above)
Comment (lakshmi-kannan) [inline scripts/st2_bootstrap_selector.sh]: For ubuntu, yes. Not sure about EL*.

File: scripts/st2_bootstrap_selector.sh [hunk 6]
```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+BASE_PATH="https://raw.githubusercontent.com/StackStorm/st2-packages/master/scripts/st2bootstrap"
+BOOTSTRAP_FILE='st2bootstrap.sh'
+
+DEBTEST=`lsb_release -a 2> /dev/null | grep Distributor | awk '{print $3}'`
```
Comment (dennybaa) [inline scripts/st2_bootstrap_selector.sh]: this is incorrect: `lsb_release` can be present on rpm distro. So I find this check not sufficient...
Why not to try `[ -f /etc/debian_version]` it seems less prone to errors.

**Comment by estee-tew** [comment]
https://stackstorm.slack.com/archives/stackstorm/p1457118046006177

**Pull request merged**